### PR TITLE
feat: add subgraph documentation

### LINKED ISSUE #1017: Add subgraph documentation

### PR DESCRIPTION
**Pull request by PaulieB14**
Closes #1017 

Added subgraph documentation with information, links, entities, and subgraph query examples.  Do I have to add anything to Docusaurus.config?

### CODE REVIEW

**Review by MantisClone — CHANGES_REQUESTED** [review]
Thanks for submitting this PR!

I requested changes on https://github.com/RequestNetwork/payments-subgraph/pull/45. Please update this PR once the changes are made.

**Comment by PaulieB14** [comment]
@MantisClone sure thing.  I was having trouble matching prettier settings on this and getting errors.

**Comment by MantisClone** [comment]
@PaulieB14 Would you mind also documenting our other two subgraphs:
* https://github.com/RequestNetwork/price-aggregators-subgraph
* https://github.com/RequestNetwork/storage-subgraph

**Comment by PaulieB14** [comment]
@MantisClone sure I can document the other two as well.  I will make revisions to this once the Schema is completed

**Comment by PaulieB14** [comment]
@MantisClone the other two subgraphs don’t seem deployed yet

**Comment by MantisClone** [comment]
@PaulieB14
The `storage` subgraphs are deployed on internal graph nodes. They are private for now. The reason why we need our own nodes is that they are using our own "private" IPFS network.

The `price aggregator` subgraphs can be found here:
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-mainnet
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-goerli
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-xdai
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-matic
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-fantom
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-bsc
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-avalanche
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-optimism
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-arbitrum-one
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-arbitrum-rinkeby
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-moonbeam
* https://thegraph.com/hosted-service/subgraph/requestnetwork/price-aggregators-rinkeby

**Comment by MantisClone** [comment]
> @MantisClone sure thing. I was having trouble matching prettier settings on this and getting errors.

`husky` runs `prettier` before each commit. You can install it by running `yarn install` in the repo root.

**Comment by PaulieB14** [comment]
> > @MantisClone sure thing. I was having trouble matching prettier settings on this and getting errors.
> 
> `husky` runs `prettier` before each commit. You can install it by running `yarn install` in the repo root.

@MantisClone sorry having difficult time on prettier errors.  Any help on this is appreciated!

File: packages/docs/package.json [hunk 1]
```diff
@@ -1,72 +1,55 @@
 {
```
Comment (MantisClone) [inline packages/docs/package.json:1]: Please revert the changes to this file. This is the `package.json` for the `docs` package, not for the top-level monorepo.

**Review by MantisClone — CHANGES_REQUESTED** [review]

File: packages/docs/docs/guides/8-request payment subgraph/0-subgraph-info.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/0-subgraph-info.md:1]: Added hosted service links

File: .husky/pre-commit [hunk 3]
```diff
@@ -2,3 +2,5 @@
 . "$(dirname -- "$0")/_/husky.sh"
 
 yarn lint-staged
+npm run recommit
+npm run recommit
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply .husky/pre-commit]: removed duplicate

File: packages/docs/docs/guides/10-request price aggregators subgraph/1-entities-aggregator.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+description: Subgraph entities and descriptions.
+---
+
+# Aggregator
+
+Description: get specific details of the aggregator
+
+| Field   | Type       | Description                 |
+| ------- | ---------- | --------------------------- |
+| id      | ID!        | The input of the aggregator |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/10-request price aggregators subgraph/1-entities-aggregator.md]: revised description

File: packages/docs/package.json (hunk 1 above)
Comment (PaulieB14) [reply packages/docs/package.json]: Sorry copied back the correct package.json for the docs

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+# Transaction
+
+Description: get specific details of the transaction
+
+| Field                | Type      | Description                                                                                  |
+| -------------------- | --------- | -------------------------------------------------------------------------------------------- |
+| id                   | ID!       | a unique ID based on IPFS hash & position of the transaction in the document                 |
+| hash                 | String!   | IPFS has                                                                                     |
+| channelId            | String!   | invoice ID for transaction                                                                   |
+| data                 | String    | request data is the current state of a request,                                              |
```
Comment (MantisClone) [inline packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:16]: Please change to "transaction data, used for clear channels"

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+| encryptedData        | String    | If the request is encrypted, the transactions are decrypted in this layer                    |
```
Comment (MantisClone) [inline packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:17]: Please change to "transaction data, used for encrypted channels"

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+| encryptionMethod     | String    | Advanced Encryption Standard (AES) or Elliptic Curve Integrated Encryption Scheme (ECIES)    |
```
Comment (MantisClone) [inline packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:18]: Please change to "Encryption methods for channel key and transactions, concatenated together"

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+| publicKeys           | [String!] | unique channel key that is shared to all the stakeholders                                    |
```
Comment (MantisClone) [inline packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:19]: Please change to "Encryption stakeholder identities included in the transaction"

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+| encryptedKeys        | [String!] | set of private keys where each is privately held by the stakeholder                          |
```
Comment (MantisClone) [inline packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:20]: Please change to "Channel key, encrypted using each stakeholder's public key"

**Review by MantisClone — CHANGES_REQUESTED** [review]
Excellent work @PaulieB14 I left some change requests.

File: .husky/pre-commit [hunk 10]
```diff
@@ -2,3 +2,5 @@
 . "$(dirname -- "$0")/_/husky.sh"
 
 yarn lint-staged
+npm run recommit
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply .husky/pre-commit]: I removed line 5

File: packages/docs/docs/guides/11-glossary.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
+---
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/11-glossary.md]: Deleted and corrected order

File: packages/docs/docs/guides/8-request payment subgraph/1-entities.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+| timestamp         | Int!        | time stamp of payment                            |
+| txHash            | Bytes!      | transaction hash of payment                      |
+| gasUsed           | BigInt!     | gas used for payment                             |
+| gasPrice          | BigInt!     | gas fee at time of payment                       |
+| amount            | BigDecimal! | amount of payment                                |
+| feeAmount         | BigDecimal  | fee charged for payment                          |
+| feeAddress        | Bytes       | address of person paying fee                     |
+| currency          | Bytes       | fiat currency payment is based on                |
+| amountInCrypto    | BigDecimal  | amount of payment in crypto                      |
+| feeAmountInCrypto | BigDecimal  | fee amoount in crypto                            |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/1-entities.md]: Corrected typo

File: packages/docs/docs/guides/8-request payment subgraph/1-entities.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+| maxRateTimespan   | Int         | maximum time span payment will reoccur           |
+
+## Escrow
+
+Description: It acts like a transparent, safe deposit box which automatically disburses the funds held inside when both parties agree that the contract has been satisfied.
+
+| Field               | Type                           | Description                                      |
+| ------------------- | ------------------------------ | ------------------------------------------------ |
+| id                  | ID!                            | hash transaction of the escrow                   |
+| contractAddress     | Bytes!                         | network contract address                         |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/1-entities.md]: Corrected line 45

File: packages/docs/docs/guides/8-request payment subgraph/1-entities.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+| paymentProxyAddress | Bytes                          | identifies a recipient and their payment account |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/1-entities.md]: corrected line 46

File: packages/docs/docs/guides/8-request payment subgraph/1-entities.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+| to                  | Bytes                          | address escrow is paid to                        |
+| events              | [`EscrowEvent!`](#escrowevent) | refer to Escrow Event                            |
+
+## EscrowEvent
+
+Description: get specific details of the Escrow Event
+
+| Field           | Type       | Description                                              |
+| --------------- | ---------- | -------------------------------------------------------- |
+| id              | ID!        | date payment set up                                      |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/1-entities.md]: corrected 65

File: packages/docs/docs/guides/8-request payment subgraph/1-entities.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+| contractAddress | Bytes!     | network contract address                                 |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/1-entities.md]: corrected 66

File: packages/docs/docs/guides/8-request payment subgraph/1-entities.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+| reference       | Bytes!     | is the reference data used to track the escrow event     |
+| escrow          | Escrow!    | smart contract that triggers when a payment is dispursed |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/1-entities.md]: corrected 68

File: packages/docs/docs/guides/8-request payment subgraph/1-entities.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+| block           | Int!       | block event took place                                   |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/8-request payment subgraph/1-entities.md]: corrected 69

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+# Transaction
+
+Description: get specific details of the transaction
+
+| Field                | Type      | Description                                                                                  |
+| -------------------- | --------- | -------------------------------------------------------------------------------------------- |
+| id                   | ID!       | a unique ID based on IPFS hash & position of the transaction in the document                 |
+| hash                 | String!   | IPFS has                                                                                     |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:14]: corrected - added h

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+| channelId            | String!   | invoice ID for transaction                                                                   |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:15]: Changed line 13

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md (hunk 5 above)
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:16]: changed line 16

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md (hunk 6 above)
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:17]: Changed line 17

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md (hunk 7 above)
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:18]: changed line 18

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md (hunk 8 above)
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:19]: changed line 19

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md (hunk 9 above)
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:20]: changed line 20

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+| data                 | String    | request data is the current state of a request,                                              |
+| encryptedData        | String    | If the request is encrypted, the transactions are decrypted in this layer                    |
+| encryptionMethod     | String    | Advanced Encryption Standard (AES) or Elliptic Curve Integrated Encryption Scheme (ECIES)    |
+| publicKeys           | [String!] | unique channel key that is shared to all the stakeholders                                    |
+| encryptedKeys        | [String!] | set of private keys where each is privately held by the stakeholder                          |
+| blockNumber          | Int!      | blocknumber of the transaction                                                               |
+| blockTimestamp       | Int!      | block timestamp of transaction                                                               |
+| transactionHash      | String!   | transaction hash of transaction                                                              |
+| smartContractAddress | String!   | smartcontract address of the transaction                                                     |
```
[the remark this answers is not in the document]
Comment (PaulieB14) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:24]: changed line 24

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (MantisClone) [reply packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:15]: Actually, the "channelId" should be "hash of the first transaction in the channel"

File: packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+description: Subgraph entities and descriptions.
+---
+
+# Transaction
+
+Description: get specific details of the transaction
+
+| Field                | Type      | Description                                                                                  |
+| -------------------- | --------- | -------------------------------------------------------------------------------------------- |
+| id                   | ID!       | hash of the first transaction in the channel                                                 |
```
Comment (MantisClone) [inline packages/docs/docs/guides/9-request storage subgraph/0-entities-storage.md:13]: Please revert this to "a unique ID based on IPFS hash & position of the transaction in the document"

**Review by MantisClone — CHANGES_REQUESTED** [review]
